csharp/source/autowrap/csharp/csharp.d

```diff
@@ -178,7 +179,11 @@ private void generateConstructors(T)(string libraryName, CSharpAggregate csagg)
     alias fqn = fullyQualifiedName!T;
 
     static if(hasMember!(T, "__ctor")) {
-        alias constructors = AliasSeq!(__traits(getOverloads, T, "__ctor"));
+        static if (__traits(getProtection, __traits(getMember, T, "__ctor")) == "export" || __traits(getProtection, __traits(getMember, T, "__ctor")) == "public") {
+            alias constructors = AliasSeq!(__traits(getOverloads, T, "__ctor"));
```

Comment:
What if one constructor is private and other is public?

Reply:
Without `__traits(getProtection, __traits(getMember, T, "__ctor")).among("export", "public")` then you get an error if the type contains a non-public constructor.